frontend/src/metabase/parameters/utils/parameter-values.js

```diff
@@ -87,47 +83,11 @@ function normalizeParameterValueForWidget(value, parameter) {
   return value;
 }
 
-// on dashboards we treat a default parameter with a set value of "" (from a query parameter)
-// to mean that the parameter value is explicitly unset.
-// this is NOT the case elsewhere (native questions, pulses) because default values are
-// automatically used in the query when unset.
-function removeAllEmptyStringParameters(pairs) {
-  return pairs
-    .map(([parameter, value]) => [parameter, value === "" ? undefined : value])
-    .filter(([parameter, value]) => hasParameterValue(value));
-}
-
-function removeUndefaultedEmptyStringParameters(pairs) {
-  return pairs
-    .map(([parameter, value]) => [
-      parameter,
-      value === "" ? parameter.default : value,
-    ])
-    .filter(([, value]) => hasParameterValue(value));
-}
-
-// when `forcefullyUnsetDefaultedParametersWithEmptyStringValue` is true, we treat defaulted parameters with an empty string value as explecitly unset.
-// This CAN'T be used with native questions because defaulted parameters are always applied on the BE when unset on the FE.
-export function getParameterValuesByIdFromQueryParams(
-  parameters,
-  queryParams,
-  metadata,
-  { forcefullyUnsetDefaultedParametersWithEmptyStringValue } = {},
-) {
-  const parameterValuePairs = parameters.map(parameter => [
-    parameter,
-    getParameterValueFromQueryParams(parameter, queryParams, metadata),
-  ]);
-
-  const transformedPairs =
-    forcefullyUnsetDefaultedParametersWithEmptyStringValue
-      ? removeAllEmptyStringParameters(parameterValuePairs)
-      : removeUndefaultedEmptyStringParameters(parameterValuePairs);
-
-  const idValuePairs = transformedPairs.map(([parameter, value]) => [
-    parameter.id,
-    value,
-  ]);
-
-  return Object.fromEntries(idValuePairs);
+export function getParameterValuesByIdFromQueryParams(parameters, queryParams) {
+  return Object.fromEntries(
+    parameters.map(parameter => [
+      parameter.id,
+      getParameterValueFromQueryParams(parameter, queryParams),
+    ]),
+  );
```

Comment:
Before parameters with "no values" wouldn't be in the returned object at all, but now there is an entry for every parameter in the `parameters` arg.

Note `getParameterValueFromQueryParams` returns a `null` value if the parameter is empty in the query params, or it's not present in the query params and there's no default.